src/ddmd/e2ir.d

```diff
@@ -4847,7 +4847,7 @@ elem *toElem(Expression e, IRState *irs)
                     if (c1)
                     {
                         // Construct: (c1 || arrayBoundsError)
-                        auto ea = buildArrayBoundsError(irs, se.loc);
+                        auto ea = buildArrayBoundsError(irs, se.loc, el_copytree(elwr2), el_copytree(eupr2), el_copytree(elen));
```

Comment:
I am no compiler expert but would `el_copytree` cause the expressions to be re-evaluated (incl. their side effects)? E.g. `int[1] a; int i = 5; a[++i] = 1;`